minari/utils.py

```diff
@@ -1,20 +1,204 @@
 from __future__ import annotations
 
+import importlib.metadata
 import os
 import warnings
 from typing import Dict, List, Optional, Union
 
 import gymnasium as gym
 import h5py
 import numpy as np
+import portion as P
 from gymnasium.envs.registration import EnvSpec
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version
 
 from minari import DataCollectorV0
 from minari.dataset.minari_dataset import MinariDataset, clear_episode_buffer
 from minari.serialization import serialize_space
 from minari.storage.datasets_root_dir import get_dataset_path
 
 
+# Use importlib due to circular import when: "from minari import __version__"
+__version__ = importlib.metadata.version("minari")
+
+
+def combine_minari_version_specifiers(specifier_set: SpecifierSet):
```

Comment:
Do we need this function? Look like the & operator of SpecifierSet returns a valid SpecifierSet that we can use to check compatibility